rackcity/views/user_views.py

```diff
@@ -442,3 +447,108 @@ def usernames(request):
         {"usernames": usernames},
         status=HTTPStatus.OK,
     )
+
+
+@api_view(['POST'])
+@permission_classes([IsAdminUser])
+def user_set_groups(request):
+    """
+    Set groups for a user.
+    """
+    data = JSONParser().parse(request)
+    if 'id' not in data:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    GenericFailure.INTERNAL.value,
+                "errors": "Must specify user id on admin permission revoke"
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    try:
+        user = User.objects.get(id=data['id'])
+    except ObjectDoesNotExist:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    "User" + GenericFailure.DOES_NOT_EXIST.value,
+                "errors": "No existing user with id=" + data['id']
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    groups_added = []
+    groups_removed = []
+    for group in GroupName:
+        group_key = group.value
+        if group_key in data:
+            if data[group_key]:
+                added = add_user_to_group(user, group)
+                if added:
+                    groups_added.append(group_key)
+            else:
+                removed = remove_user_from_group(user, group)
+                if removed:
+                    groups_removed.append(group_key)
+    success_message = ""
+    if len(groups_added) > 0:
+        success_message += \
+            ("User added to groups: " + ", ".join(groups_added) + ". ")
+    if len(groups_removed) > 0:
+        success_message += \
+            ("User removed from groups: " + ", ".join(groups_removed) + ". ")
+    if len(groups_added) == 0 and len(groups_removed) == 0:
+        success_message += \
+            "User's groups were not changed."
+    return JsonResponse(
+        {"success_message": Status.SUCCESS.value + success_message},
+        status=HTTPStatus.OK,
+    )
+
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def user_get_groups(request):
```

Comment:
endpoint to get the groups that a particular user belongs to (returned as a list of strings of group names)